pylint/checkers/refactoring.py

```diff
@@ -261,13 +266,48 @@ def _check_redefined_argument_from_local(self, name_node):
                                  args=(name_node.name, ))
 
     @utils.check_messages('redefined-argument-from-local',
-                          'too-many-nested-blocks')
+                          'too-many-nested-blocks',
+                          'mutable-sequence-modified-in-loop')
     def visit_for(self, node):
         self._check_nested_blocks(node)
 
         for name in node.target.nodes_of_class(astroid.AssignName):
             self._check_redefined_argument_from_local(name)
 
+        if self._is_iterating_mutable_sequence(node):
+            parts = node.nodes_of_class(astroid.Call)
+            if isinstance(parts, collections.Iterable):
+                for part in parts:
+                    if self._check_part_is_call_method_elimination(node.iter,
+                                                                   part.func):
+                        self.add_message('mutable-sequence-modified-in-loop',
+                                         node=part,
+                                         args=(node.iter.name))
+
+    @staticmethod
+    def _is_iterating_mutable_sequence(node):
+        """Check if node.iter is astroid.util.Uninferable o
+        astroid.node_classes.List instance and if node.body contains records"""
+        inferred = utils.safe_infer(node.iter)
+        return (node.body and isinstance(node.iter, astroid.Name) and
+                (inferred is astroid.util.Uninferable or
```

Comment:
@rogalski 
I don't understand
Are you referring I cannot assume this?
```pytho
293                (inferred is astroid.util.Uninferable or
```
What other option can I use?